pandas/core/indexes/base.py

```diff
@@ -3549,8 +3549,12 @@ def _join_multi(self, other, how, return_indexers=True):
             ldrop_names = list(self_names - overlap)
             rdrop_names = list(other_names - overlap)
 
-            self_jnlevels = self.droplevel(ldrop_names)
-            other_jnlevels = other.droplevel(rdrop_names)
+            if len(ldrop_names + rdrop_names) == 0:  # if only the order differs
```

Comment:
rather than add a special case, is there an issue droplevel?

Reply:
I'm sorry, "issue droplevel"?

Reply:
I am not sure what you are trying to fix here. is there somthing that is actually wrong with droplevel instead of adding a special case here

Reply:
nothing is wrong with droplevel, it just gets called twice with empty lists if ```len(ldrop_names + rdrop_names) == 0``` is true, which seems pointless. and then because this doesn't change anything, it ends up in an infinite recursion loop. this can be stopped with checking for this condition, and the reordering of the names makes the join work.

Reply:
ok, this is not idiomatic, use
```
if not len(....)
 ....
```
put he comment on the line above